### PR TITLE
[SofaGeneralEngine] Fix BarycentricMapperEngine parse() function

### PR DESCRIPTION
**Pull request by fredroy**
Bug introduced in #1487, with an infinite loop when calling parse()
(Fix timeout with MeshBarycentricMapperEngine.scn on the CI)

______________________________________________________

This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: modules/SofaGeneralEngine/MeshBarycentricMapperEngine.h [hunk 1]
```diff
@@ -83,7 +83,7 @@ class MeshBarycentricMapperEngine : public core::DataEngine
         {
             msg_warning() << "input data 'InputMeshName' changed for 'topology', please update your scene (see PR#1487)";
         }
-        MeshBarycentricMapperEngine::parse(arg);
+        core::DataEngine::parse(arg);
```
Comment (epernod) [inline modules/SofaGeneralEngine/MeshBarycentricMapperEngine.h:86]: does it make a diff to do that call or ```Inherit1::parse(arg);``` ?

Comment (fredroy) [reply modules/SofaGeneralEngine/MeshBarycentricMapperEngine.h:86]: Inherit1 ?
You mean to make a typedef Inherit1 = core::DataEngine and calling it ?

Comment (epernod) [reply modules/SofaGeneralEngine/MeshBarycentricMapperEngine.h:86]: my bad I thought it was a smarter implementation. 
ok for me

**Review by epernod — APPROVED** [review]

**Pull request merged**